src/ml/algorithm_analysis.py

```diff
@@ -0,0 +1,131 @@
+import pandas as pd
+from typing import Iterable
+from pathlib import Path
+import os
+
+import matplotlib.pyplot as plt
+from sklearn.decomposition import PCA
+from sklearn.preprocessing import StandardScaler
+import seaborn as sns
+
+from sklearn.cluster import AgglomerativeClustering
+from scipy.cluster.hierarchy import dendrogram
+import numpy as np
+
+import argparse
+
+def is_file_empty(filepath):
```

Comment:
We'll need to use this function when loading networks, or if we switch to networkx perhaps it handles loading empty graphs.